digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/PostGISAssetDao.scala

```diff
@@ -39,13 +103,50 @@ class PostGISAssetDao {
   }
 
   /**
-    * When invoked will expire assets by Id.
+    * When invoked will expire given assets by Id.
     * It is required that the invoker takes care of the transaction.
     *
     * @param id Represets the id of the asset
     */
-  def expireAssetsById (id: Long): Unit = {
-    sqlu"update asset set valid_to = INTERVAL'1 SECOND' where id = $id".execute
+  def expireAssetsById (ids: Set[Long], userName: String): Unit = {
+    sqlu""" update asset
+            set valid_to = now(),
+            modified_by = ${userName}
+            where id in(#${ids.mkString(",")})""".execute
+
+  }
+
+  def getAssetsOnExpiredRoadLinksById(ids: Set[Long]):Seq[AssetOnExpiredRoadLink] = {
+    MassQuery.withIds(ids) { idTableName =>
+      sql"""
+       SELECT
+        a.id,
+        lp.link_id,
+        lp.start_measure,
+        lp.end_measure,
+        a.geometry,
+        kr.shape AS line_geometry,
+        lp.side_code,
+        a.bearing,
+        CASE
+          WHEN lp.start_measure IS NOT NULL
+            AND lp.end_measure IS NOT NULL
+          THEN ST_LineMerge(
+            ST_LocateBetween(
+              kr.shape,
+              lp.start_measure / ST_Length(kr.shape),
```

Comment:
```suggestion
              lp.start_measure / kr.geometrylength,
```

Reply:
Tää koska DR lasketaan m-arvot lähes aina geometrylength avulla. Varmaan jossakin tehdään toisella tavalla mut 99% geometrylength avulla. ST_Length ja geometrylength pitäisi kyl palauttaa saman vastauksen mutta näin ei aina käy.
Parempi käyttää geometrylength niin saadaan taatusti oikea vastaus.